lamindb_setup/_load_instance.py

```diff
@@ -132,6 +139,32 @@ def load(
     return None
 
 
+def get_db_from_vault(instance_result):
+    from lamin_vault.client._create_vault_client import (
+        create_vault_authenticated_client,
+    )
+
+    try:
+        vault_client = create_vault_authenticated_client(
+            access_token=settings.user.access_token,
+        )
+        config_db_exists = connection_config_db_exists(
+            vault_client, instance_result["id"]
+        )
+        if config_db_exists:
+            return get_db_from_vault_base(
+                vault_client=vault_client,
+                scheme=instance_result["db_scheme"],
+                host=instance_result["db_host"],
+                port=instance_result["db_port"],
+                name=instance_result["db_database"],
+                role=f'{instance_result["id"]}-{settings.user.uuid}-db',
+            )
+    except Exception:
```

Comment:
This exception seems like a bad idea. Wouldn't we rather want to throw an error? If we enabled vault and we can't get credentials, we'll error downstream of this call, won't we?

Reply:
I don't know, I wanted to avoid introducing a new SPOF with this as it requires making a call to `laminhub-rest` server. 

Here, in case in failure, we will just fallback to the logic we have without vault. 

Do you think it's better to raise an error and to make load fails in that case?

Reply:
Oh, I didn't parse this!

I'd still actually raise the error. We now want to test using lamindb-setup in vault mode and I fear people won't even notice if things go wrong if there is no clear error (or, they notice in a weird unexpected downstream way).

If you really think you want this magic under-the-hood, could you add a comment that this fallback to the original logic?